version: '3.8'
services:
  postgres_db_stock_management:
    container_name: postgres_db_stock_management
    image: postgres:14-alpine3.18
    restart: always
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: password
      POSTGRES_DB: stock_management
    ports:
      - "5432:5432"
    volumes:
      - volume_postgres_db_stock_management:/var/lib/postgresql/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: "pg_isready -U postgres | grep 'accepting connections' || exit 1"
      interval: 2s
      timeout: 20s
      retries: 10
volumes:
  volume_postgres_db_stock_management:
    external: false